x-pack/plugins/ml/public/application/trained_models/models_management/start_deployment_setup.tsx

```diff
@@ -0,0 +1,226 @@
+ */
+
+import React, { FC, useState, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiForm,
+  EuiButtonGroup,
+  EuiFormRow,
+  EuiFieldNumber,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+} from '@elastic/eui';
+import { toMountPoint, wrapWithTheme } from '@kbn/kibana-react-plugin/public';
+import type { Observable } from 'rxjs';
+import type { CoreTheme, OverlayStart } from '@kbn/core/public';
+
+interface StartDeploymentSetup {
+  config: ThreadingParams;
+  onConfigChange: (config: ThreadingParams) => void;
+}
+
+export interface ThreadingParams {
+  numOfAllocations: number;
+  threadsPerAllocations: number;
+}
+
+const THREADS_MAX_EXPONENT = 6;
+
+/**
+ * Form for setting threading params.
+ */
+export const StartDeploymentSetup: FC<StartDeploymentSetup> = ({ config, onConfigChange }) => {
+  const numOfAllocation = config.numOfAllocations;
+  const threadsPerAllocations = config.threadsPerAllocations;
+
+  const threadsPerAllocationsOptions = useMemo(
+    () =>
+      new Array(THREADS_MAX_EXPONENT).fill(null).map((v, i) => {
+        const value = Math.pow(2, i);
+        const id = value.toString();
+
+        return {
+          id,
+          label: id,
+          value,
+        };
+      }),
+    []
+  );
+
+  const toggleIdSelected = threadsPerAllocationsOptions.find(
+    (v) => v.value === threadsPerAllocations
+  )!.id;
+
+  return (
+    <EuiForm component={'form'} id={'startDeploymentForm'}>
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsLabel"
+            defaultMessage="Number of allocations"
+          />
+        }
+        hasChildLabel={false}
+        helpText={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsHelp"
+            defaultMessage="Each allocation means the model gets another thread for executing parallel inference requests"
+          />
+        }
+      >
+        <EuiFieldNumber
+          min={1}
+          name={'numOfAllocations'}
+          value={numOfAllocation}
+          onChange={(event) => {
+            onConfigChange({ ...config, numOfAllocations: Number(event.target.value) });
+          }}
+        />
+      </EuiFormRow>
+
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationLabel"
+            defaultMessage="Threads per allocation"
+          />
+        }
+        hasChildLabel={false}
+        helpText={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationHelp"
+            defaultMessage="Each allocation is may be using a number of threads to parallelize each individual inference request"
```

Comment:
As above, this info seems very low-level. Looking at the docs, how about something simpler like

```
The number of threads used by each model allocation during inference. This generally increases the inference speed. 
```

with a link to the docs for more info.